sycl/source/detail/adapter.hpp

```diff
@@ -35,6 +34,7 @@
 
 namespace sycl {
 inline namespace _V1 {
+enum class backend : char;
```

Comment:
Forward declarations of enums classes make me slightly uneasy since the storage types can end up mismatching (which can't happen with a struct forward declaration for example). I've had to track down nasty bugs due to this in the past.

Do we have a way to ensure that if the complete declaration changes type, we know to update all of these forward decls?

Reply:
> Do we have a way to ensure that if the complete declaration changes type, we know to update all of these forward decls?

I actually had to type the exact type here, or otherwise compiler complained that forward declaration doesn't match the actual definition. From that point of view, I think it should be safe for us to say that such mechanism exists.

Note that I was using `clang` as compiler and it could have been a diagnostic specific to it

Reply:
If clang is catching it then I think that's good enough. Thanks